ccos_image.c

```diff
@@ -42,66 +42,65 @@ int ccos_check_image(const uint8_t* file_data) {
   return 0;
 }
 
-uint16_t ccos_file_id(ccos_inode_t* inode) {
+uint16_t ccos_file_id(const ccos_inode_t* inode) {
   return inode->header.file_id;
 }
 
-version_t ccos_get_file_version(ccos_inode_t* file) {
-  uint8_t major = file->version_major;
-  uint8_t minor = file->version_minor;
-  uint8_t patch = file->version_patch;
+version_t ccos_get_file_version(const ccos_inode_t* file) {
+  uint8_t major = file->desc.version_major;
+  uint8_t minor = file->desc.version_minor;
+  uint8_t patch = file->desc.version_patch;
   version_t version = {major, minor, patch};
   return version;
 }
 
-int ccos_set_file_version(ccos_inode_t* file, version_t new_version) {
-  if (!is_root_dir(file)) {
-    file->version_major = new_version.major;
-    file->version_minor = new_version.minor;
-    file->version_patch = new_version.patch;
-    update_inode_checksums(file);
-    return 0;
-  }
+int ccos_set_file_version(ccfs_handle ctx, ccos_inode_t* file, version_t new_version) {
+  if (is_root_dir(file)) return -1;
```

Comment:
While not related to the PR in question, early return is OK here. I prefer code style without one-liner blocks. Please add curly braces.